rplidar_test.go

```diff
@@ -0,0 +1,226 @@
+package rplidar
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/golang/geo/r3"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/test"
+)
+
+func TestValidate(t *testing.T) {
+	t.Run("min range is zero", func(t *testing.T) {
+		cfg := Config{
+			MinRangeMM: 0,
+		}
+
+		deps, err := cfg.Validate("")
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(deps), test.ShouldEqual, 0)
+	})
+	t.Run("min range is greater than zero", func(t *testing.T) {
+		cfg := Config{
+			MinRangeMM: 1,
+		}
+
+		deps, err := cfg.Validate("")
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, len(deps), test.ShouldEqual, 0)
+	})
+	t.Run("min range is less than zero", func(t *testing.T) {
+		cfg := Config{
+			MinRangeMM: -1,
+		}
+
+		deps, err := cfg.Validate("")
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldEqual, "min_range must be positive")
+		test.That(t, len(deps), test.ShouldEqual, 0)
+	})
+}
+
+func TestScan(t *testing.T) {
```

Comment:
[note] if the 'valid rplidar driver that returns all points at origin' test is removed (see other comment for details), then I can/will revert from map based struct